src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,80 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private static final Class<ProductParser> productParser = ProductParser.class;
+    private static final Class<ProductParserImpl> productParserImpl = ProductParserImpl.class;
+    private static final Class<FileReaderService> fileReaderService = FileReaderService.class;
+    private static final Class<FileReaderServiceImpl> fileReaderServiceImpl
+            = FileReaderServiceImpl.class;
+    private static final Class<ProductService> productService = ProductService.class;
+    private static final Class<ProductServiceImpl> productServiceImpl = ProductServiceImpl.class;
+    private Map<Class<?>, Object> instances = new HashMap<>();
```

Comment:
```suggestion
    private final Map<Class<?>, Object> instances;
```
and init it in constructor using Map.of